src/test/javascript/spec/component/shared/sidebar/sidebar.helper.spec.ts

```diff
@@ -0,0 +1,338 @@
+            ],
+        });
+    });
+
+    it('should filter duplicates', () => {
+        const sidebarData: SidebarData = {
+            groupByCategory: true,
+            ungroupedData: [SIDEBAR_CARD_ELEMENT_1, SIDEBAR_CARD_ELEMENT_4],
+        };
+
+        const exerciseCategories = getExerciseCategoryFilterOptions(undefined, sidebarData);
+        expect(exerciseCategories).toEqual({
+            isDisplayed: true,
+            options: [
+                { category: new ExerciseCategory('#691b0b', 'category1'), searched: false },
+                { category: new ExerciseCategory('#1b97ca', 'category2'), searched: false },
+                { category: new ExerciseCategory('#1b97ca', 'category8'), searched: false },
+            ],
+        });
+    });
+
+    it('should sort categories alphanumerical', () => {
+        const sidebarData: SidebarData = {
+            groupByCategory: true,
+            ungroupedData: [SIDEBAR_CARD_ELEMENT_2, SIDEBAR_CARD_ELEMENT_1],
+        };
+
+        const exerciseCategories = getExerciseCategoryFilterOptions(undefined, sidebarData);
+        expect(exerciseCategories).toEqual({
+            isDisplayed: true,
+            options: [
+                { category: new ExerciseCategory('#691b0b', 'category1'), searched: false },
+                { category: new ExerciseCategory('#1b97ca', 'category2'), searched: false },
+                { category: new ExerciseCategory('#0d3cc2', 'category3'), searched: false },
+                { category: new ExerciseCategory('#6ae8ac', 'category4'), searched: false },
+            ],
+        });
+    });
+
+    it('should not be displayed if no categories are available', () => {
+        const sidebarData: SidebarData = {
+            groupByCategory: true,
+            ungroupedData: [SIDEBAR_CARD_ELEMENT_5],
+        };
+
+        const exerciseCategories = getExerciseCategoryFilterOptions(undefined, sidebarData);
+        expect(exerciseCategories).toEqual({
+            isDisplayed: false,
+            options: [],
+        });
+    });
+
+    it('should directly return if already initialized', () => {
+        const sidebarData: SidebarData = {
+            groupByCategory: true,
+            ungroupedData: [SIDEBAR_CARD_ELEMENT_1, SIDEBAR_CARD_ELEMENT_2],
+        };
+
+        const exerciseCategories = getExerciseCategoryFilterOptions(
+            {
+                categoryFilter: {
+                    isDisplayed: false,
+                    options: [],
+                },
+            },
+            sidebarData,
+        );
+        expect(exerciseCategories).toEqual({
+            isDisplayed: false,
+            options: [],
+        });
+    });
+});
+
+describe('getExerciseTypeFilterOptions', () => {
+    it('should return present exercise types and sort them properly (same order as instructor creation)', () => {
+        const sidebarData: SidebarData = {
```

Comment:
Is it possible to generalize this initialization as it is used frequently? If not it is also fine, no strong opinion on this.

Reply:
This is only the same object for one test in `getExerciseTypeFilterOptions` and `getExerciseDifficultyFilterOptions` - I think that it is a better approach to keep it separated to reduce the possibility of side effects when adjusting the tests

My opinion is not very strong here. Do you disagree with that reasoning?